feature/experimental/policy/otg_tests/prefix_set_test/prefix_set_test.go

```diff
@@ -100,45 +108,47 @@ func TestPrefixSet(t *testing.T) {
 	dut := ondatra.DUT(t, "dut")
-	ctx := context.Background()
-	cli := dut.RawAPIs().CLI(t)
-
-	switch dut.Vendor() {
-	case ondatra.ARISTA:
-		helpers.GnmiCLIConfig(t, dut, `
-			ip prefix-list TAG_3_IPV4
-				seq 10 permit 10.240.31.48/28
-				seq 20 permit 10.244.187.32/28
-				seq 30 permit 173.36.128.0/20
-				seq 40 permit 173.37.128.0/20
-				seq 50 permit 173.38.128.0/20
-				seq 60 permit 173.39.128.0/20
-				seq 70 permit 173.40.128.0/20
-				seq 80 permit 173.41.128.0/20
-				seq 90 permit 173.42.128.0/20
-				seq 100 permit 173.43.128.0/20
-			`,
-		)
-		helpers.GnmiCLIConfig(t, dut, `
-			management api gnmi
-				transport grpc default
-				operation set persistence
-			`,
-		)
-		cmd := "agent Octa terminate"
-		res, err := cli.RunCommand(ctx, "agent Octa terminate")
-		if err != nil {
-			t.Errorf("error executing command %q:\n%v", cmd, err)
-		}
-		if res.Error() != "" {
-			t.Errorf("error executing command %q:\n%v", cmd, res.Error())
-		}
-	}
 
 	dutOcRoot := &oc.Root{}
 	rp := dutOcRoot.GetOrCreateRoutingPolicy()
 	ds := rp.GetOrCreateDefinedSets()
-	v4PrefixSet := ds.GetOrCreatePrefixSet("TAG_3_IPV4")
+	v4PrefixSet := ds.GetOrCreatePrefixSet(tag3IPv4)
+	v4PrefixSet.GetOrCreatePrefix("10.240.31.48/28", mskLen)
+	v4PrefixSet.GetOrCreatePrefix("10.244.187.32/28", mskLen)
+	v4PrefixSet.GetOrCreatePrefix("173.36.128.0/20", mskLen)
+	v4PrefixSet.GetOrCreatePrefix("173.37.128.0/20", mskLen)
+	v4PrefixSet.GetOrCreatePrefix("173.38.128.0/20", mskLen)
+	v4PrefixSet.GetOrCreatePrefix("173.39.128.0/20", mskLen)
+	v4PrefixSet.GetOrCreatePrefix("173.40.128.0/20", mskLen)
+	v4PrefixSet.GetOrCreatePrefix("173.41.128.0/20", mskLen)
+	v4PrefixSet.GetOrCreatePrefix("173.42.128.0/20", mskLen)
+	v4PrefixSet.GetOrCreatePrefix("173.43.128.0/20", mskLen)
+	gnmi.Update(t, dut, gnmi.OC().RoutingPolicy().DefinedSets().PrefixSet(tag3IPv4).Config(), v4PrefixSet)
+	prefixSet := gnmi.Get[*oc.RoutingPolicy_DefinedSets_PrefixSet](t, dut, gnmi.OC().RoutingPolicy().DefinedSets().PrefixSet(tag3IPv4).State())
+	if len(prefixSet.Prefix) != 10 {
+		t.Errorf("Prefix set has %v prefixes, want 10", len(prefixSet.Prefix))
+	}
+
+	gnoiClient := dut.RawAPIs().GNOI(t)
+	if _, ok := processName[dut.Vendor()]; !ok {
+		t.Fatalf("Please add support for vendor %v in var processName", dut.Vendor())
+	}
+	killProcessRequest := &spb.KillProcessRequest{
+		Signal:  spb.KillProcessRequest_SIGNAL_TERM,
+		Name:    processName[dut.Vendor()],
+		Pid:     findProcessByName(context.Background(), t, dut, processName[dut.Vendor()]),
+		Restart: true,
+	}
+	processKillResponse, err := gnoiClient.System().KillProcess(context.Background(), killProcessRequest)
+	if err != nil {
+		t.Fatalf("Failed to restart process %v with unexpected err: %v", processName[dut.Vendor()], err)
+	}
+
+	t.Logf("gnoiClient.System().KillProcess() response: %v, err: %v", processKillResponse, err)
+	t.Logf("Wait 60 seconds for process to restart ...")
+	time.Sleep(60 * time.Second)
```

Comment:
maybe a function in `/internal/gnoi` such as killRoutingProcess with a parameter if the process should be checked for restart?  

Better to Subscribe TARGET_DEFINED to /system/processes to see when the process restarts, and add a timeout for how long to wait.